stack-client: init at 5.3.1

timoteuszelle opened the pull request:
Adds the Stack client, a command-line tool and GUI for TransIP's Stack cloud storage service.

Stack is a cloud storage service provided by TransIP (https://www.transip.nl/stack/), 
offering secure file storage and synchronization capabilities.

This PR adds the command-line client and GUI client that allows users to interact with Stack from their terminal.

- [x] Tested build on x86_64-linux
- [x] Package is fetchable and builds successfully
- [x] Added relevant meta attributes (description, homepage, license)

ethancedwards8 commented on pkgs/by-name/st/stack-client/kdsingleapplication.nix:
```diff
@@ -0,0 +1,34 @@
+  qt6,
+}:
+
+stdenv.mkDerivation rec {
```
please use the new finalAttrs pattern

ethancedwards8 commented on pkgs/by-name/st/stack-client/kdsingleapplication.nix:
```diff
@@ -0,0 +1,34 @@
+
+  src = fetchFromGitHub {
+    owner = "KDAB";
+    repo = pname;
```
using pname like this is discouraged nowadays, just put the package name

ethancedwards8 commented on pkgs/by-name/st/stack-client/kdsingleapplication.nix:
```diff
@@ -0,0 +1,34 @@
+  cmakeFlags = [ "-DKDSingleApplication_QT6=true" ];
+  dontWrapQtApps = true;
+
+  meta = with lib; {
```
remove with lib;

ethancedwards8 commented on pkgs/by-name/st/stack-client/kdsingleapplication.nix:
```diff
@@ -0,0 +1,34 @@
+{
```
this can probably be its own package in its own directory

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+let
+  kdsingleapplication = callPackage ./kdsingleapplication.nix { };
+
+  libre-graph-api = stdenv.mkDerivation rec {
+    pname = "libre-graph-api-cpp-qt-client";
+    version = "1.0.4";
+
+    src = fetchFromGitHub {
+      owner = "owncloud";
+      repo = "libre-graph-api-cpp-qt-client";
+      rev = "v${version}";
+      sha256 = "wbdamPi2XSLWeprrYZtBUDH1A2gdp6/5geFZv+ZqSWk=";
+    };
+
+    preConfigure = "cd client";
+
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+      qt6.wrapQtAppsHook
+    ];
+
+    buildInputs = [
+      qt6.qtbase
+      qt6.qttools
+    ];
+
+    meta = with lib; {
+      description = "C++ Qt client library for Libre Graph API";
+      homepage = "https://github.com/owncloud/libre-graph-api-cpp-qt-client";
+      license = licenses.asl20;
+      platforms = platforms.linux;
+    };
```
this should be its own package in its own directory

timoteuszelle replied:
I did that initially, it didn't pass the automated checks?

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+      qt6.qttools
+    ];
+
+    meta = with lib; {
```
remove with lib;

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+    src = fetchFromGitHub {
+      owner = "owncloud";
+      repo = "libre-graph-api-cpp-qt-client";
+      rev = "v${version}";
```
use tag

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+let
+  kdsingleapplication = callPackage ./kdsingleapplication.nix { };
+
+  libre-graph-api = stdenv.mkDerivation rec {
```
use finalAttrs pattern

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+      sha256 = "wbdamPi2XSLWeprrYZtBUDH1A2gdp6/5geFZv+ZqSWk=";
+    };
+
+    preConfigure = "cd client";
```
sourceRoot is probably the preferred way to do this

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+    };
+  };
+in
+stdenv.mkDerivation rec {
```
please use the new finalAttrs pattern

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+  version = "latest";
+
+  src = fetchurl {
+    url = "https://filehosting-client.transip.nl/packages/stack-source-latest.tar.gz";
```
this probalby isn't a stable url. Can you find a versioned url?

timoteuszelle replied:
I have been using it as appimage for over a year, the source has not moved, this is a hosting/cloud provider in the netherlands after al. https://filehosting-client.transip.nl/stack/linux

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+    kdePackages.kcoreaddons
+  ];
+
+  meta = with lib; {
```
remove with lib;

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+  meta = with lib; {
+    description = "Stack Client for TransIP";
+    homepage = "https://www.transip.nl/stack";
+    license = licenses.unfree;
```
needs maintainers

ethancedwards8 commented on pkgs/by-name/st/stack-client/package.nix:
```diff
@@ -0,0 +1,82 @@
+    meta = with lib; {
+      description = "C++ Qt client library for Libre Graph API";
+      homepage = "https://github.com/owncloud/libre-graph-api-cpp-qt-client";
+      license = licenses.asl20;
```
needs maintainers

ethancedwards8 reviewed (changes requested):
Some minor things. Thank you for your contribution!

timoteuszelle commented:
The rest I will look at and make changes as suggested.

Commit:
stack-client: init at 5.3.1-20240731

Commit:
maintainers: add timoteuszelle

hellwolf reviewed (approved):
thanks, lgtm!

timoteuszelle committed:
Merge branch 'NixOS:master' into stack-client-package

timoteuszelle commented:
> Do not modify irrelevant files and codes

Thank you for your response, can you mark files/lines you are referring too?

timoteuszelle commented on maintainers/maintainer-list.nix:
```diff
@@ -26002,7 +26002,12 @@
     github = "timor";
     githubId = 174156;
-    name = "timor";
+    name = "Timor Kellers";
```
I’ve reverted unrelated edits and added my own maintainer handle in a separate commit, as requested.

timoteuszelle commented:
> It is better to submit using the email address associated with your GitHub account. Names like NixOS User are not good, it is better to use your GitHub username. Adding maintainers and adding maintainers for other packages should not be done in the same commit. Keep the commit history clean and do not introduce "Merge branch 'NixOS:master' into stack-client-package"

Thanks for the review. I’ve reverted unrelated edits and split commits as requested.
Changes:
•  Package moved to by-name: pkgs/by-name/st/stack-client/package.nix
•  Version pinned to v5.3.1-20240731 with fixed-output hash; sourceRoot = "client"
•  Reuse existing deps (qt6, extra-cmake-modules, qtkeychain, sqlite, zlib, libsecret, libre-graph-api-cpp-qt-client, kdsingleapplication)
•  Updater and tests disabled at configure-time
•  meta: mainProgram = "stack", license = GPL-2.0-or-later, platforms.linux
•  Maintainer entry added (timoteuszelle) in a separate commit
libre-graph-api-cpp-qt-client is unchanged now. Commits are authored with my GitHub identity and history is linear (no merges). It builds locally:
•  nix build .#stack-client
•  ./result/bin/stack --version

qzylinra commented:
Sorry to bother you. You can work on this PR at your leisure. Please do not mark unresolved issues as resolved prematurely.
